src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/FunctionsTests.cs

```diff
@@ -650,6 +651,7 @@ private void TestUdf()
             Udf<int, int>((arg) => arg);
             Udf<long, long>((arg) => arg);
             Udf<short, short>((arg) => arg);
+            Udf<DateTime, Date>((arg) => new Date(arg));
```

Comment:
I thought about this a bit more and chaining will not work if UDF takes `DateTime` instead of `Date`. Can we register a custom unpickler for `datetime.date`? Check here: https://github.com/irmen/Pyrolite/blob/pyro4-legacy/dotnet/Razorvine.Pyrolite/Pyrolite/Pickle/Unpickler.cs#L45

Reply:
Resolved as per offline discussion. The new changes is in.